### PR TITLE
Start cosmos-sdk v0.47 integration

### PR DESCRIPTION
**Pull request by alpe**
Start integration with current dev branches
* [Cosmos SDK v0.47 branch ](https://github.com/cosmos/cosmos-sdk/tree/release/v0.47.x)
* [IBC-go branch ](https://github.com/cosmos/ibc-go/tree/marko/0.47-test)

With this PR the [inter-tx](https://github.com/cosmos/interchain-accounts-demo) authentication module for ICA is deactivated. There is no SDK 47 version available, yet

Adds new modules:
* group
* nft
 
### Left TODO:
- [x] fix linter issues
- [x] fix race condition in tests
- [x] fix simulations

Note to reviewer: please use squash and merge

### CODE REVIEW

**Comment by alpe** [comment]
@Mergifyio rebase

**Comment by faddat** [comment]
hey :)

do you have any thoughts on adopting the dependency injection style, or on the dependency injection style itself?

**Comment by alpe** [comment]
@faddat 
> do you have any thoughts on adopting the dependency injection style, or on the dependency injection style itself?

I want to avoid additional risks and complexity at this stage and focus on integration and support for new modules. Once things are stable and run on production, we can provide both templates in /app for some time to let chain developers decide.

**Comment by faddat** [comment]
ah, super.  yeah I am not 100% on weather or not it is the right direction, either. 

Thanks for your work on this.

File: x/wasm/client/proposal_handler.go [hunk 1]
```diff
@@ -4,22 +4,21 @@ import (
 	govclient "github.com/cosmos/cosmos-sdk/x/gov/client"
 
 	"github.com/CosmWasm/wasmd/x/wasm/client/cli"
-	"github.com/CosmWasm/wasmd/x/wasm/client/rest" //nolint:staticcheck
+	//nolint:staticcheck
```
Comment (pinosu) [inline x/wasm/client/proposal_handler.go:7]: can be removed

File: benchmarks/app_test.go [hunk 2]
```diff
@@ -2,65 +2,119 @@ package benchmarks
...
+	wasmApp := app.NewWasmApp(log.NewTMLogger(log.NewSyncWriter(os.Stdout)), db, nil, true, wasm.EnableAllProposals, simtestutil.EmptyAppOptions{}, nil)
+
 	if withGenesis {
-		return wasmApp, app.NewDefaultGenesisState()
+		return wasmApp, app.NewDefaultGenesisState(wasmApp.AppCodec())
 	}
 	return wasmApp, app.GenesisState{}
 }
 
 // SetupWithGenesisAccounts initializes a new WasmApp with the provided genesis
```
Comment (pinosu) [inline benchmarks/app_test.go:46]: it should be `// SetupWithGenesisAccountsAndValSet`

**Review by pinosu — APPROVED** [review]
Great job! 🥇 I added some minor comments.
On top of this cosmos-sdk proto dependency needs to be upgraded to 0.47.x when available. I have created a new issue for it #1148

**Pull request merged**